packages/docker/src/plugins/plugin.spec.ts

```diff
@@ -0,0 +1,652 @@
+import { CreateNodesContext } from '@nx/devkit';
+import { TempFs } from 'nx/src/internal-testing-utils/temp-fs';
+import { createNodesV2 } from './plugin';
+
+jest.mock('nx/src/utils/cache-directory', () => ({
+  ...jest.requireActual('nx/src/utils/cache-directory'),
+  workspaceDataDirectory: 'tmp/project-graph-cache',
+}));
+
+describe('@nx/docker', () => {
+  let createNodesFunction = createNodesV2[1];
+  let context: CreateNodesContext;
+  let tempFs: TempFs;
+  let cwd: string;
+
+  beforeEach(async () => {
+    tempFs = new TempFs('test');
+    cwd = process.cwd();
+    process.chdir(tempFs.tempDir);
+    context = {
+      nxJsonConfiguration: {
+        namedInputs: {
+          default: ['{projectRoot}/**/*'],
+          production: ['!{projectRoot}/**/*.spec.ts'],
+        },
+      },
+      workspaceRoot: tempFs.tempDir,
+      configFiles: [],
+    };
+  });
+
+  afterEach(() => {
+    jest.resetModules();
+    tempFs.cleanup();
+    process.chdir(cwd);
+  });
+
+  it('should create nodes based on Dockerfile', async () => {
+    await tempFs.createFiles({
+      'proj/Dockerfile': 'FROM node:18',
+      'proj/project.json': '{}',
+    });
+
+    const results = await createNodesFunction(['proj/Dockerfile'], {}, context);
+
+    expect(results).toMatchInlineSnapshot(`
+      [
+        [
+          "proj/Dockerfile",
+          {
+            "projects": {
+              "proj": {
+                "metadata": {},
+                "root": "proj",
+                "targets": {
+                  "docker:build": {
+                    "command": "docker build .",
+                    "inputs": [
+                      "production",
+                      "^production",
+                    ],
+                    "metadata": {
+                      "description": "Run Docker build",
+                      "help": {
+                        "command": "docker build --help",
+                        "example": {
+                          "options": {
+                            "cache-from": "type=s3,region=eu-west-1,bucket=mybucket .",
+                            "cache-to": "type=s3,region=eu-west-1,bucket=mybucket .",
+                          },
+                        },
+                      },
+                      "technologies": [
+                        "docker",
```

Comment:
@Coly010 

Can we add an icon for our project details view please?

Can you work with the Nx Cloud team to get an icon for us please?